### PR TITLE
[LayoutDOM]: fix test command and add reusable report service

### PR DESCRIPTION
**Pull request by sergii-nosachenko**
## Description

This PR resolves the issue where Cypress tests in `layoutDOM` projects were not executed because the server was not running. Additionally, it introduces an improved reporting feature by introducing a new reusable Report service.

- version will be updated and published after review

### Screenshots

Before/after demo

https://github.com/mate-academy/fed/assets/54940595/55bf0bf2-3819-4afd-bf1f-b846030e1553

### CODE REVIEW

File: mate-scripts/src/services/Report.service.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+    await execBashCodeAsync(
+      `${this.binDir}mochawesome-merge "${partsGlob}" > ${this.mergedReport}`,
+      { shouldBindStdout: this.shouldShowLogs },
+    );
+  }
+
+  private async generateHtmlReport() {
+    await execBashCodeAsync(
+      `${this.binDir}marge -o ${this.reportsDir} ${this.mergedReport}`,
```
Comment (mgrinko) [inline mate-scripts/src/services/Report.service.ts:141]: what is `${this.binDir}marge`?
Shouldn't it be `merge`?

File: mate-scripts/src/services/Report.service.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+    this.isReporterStarted = true;
+  }
+
+  async runAfterTests() {
+    if (!this.isReporterStarted) {
+      this.log('REPORTER WAS NOT STARTED. SKIP REPORTS PREPARING');
+    }
+
+    let hasBeenReportsCreated = false;
```
Comment (mgrinko) [inline mate-scripts/src/services/Report.service.ts:43]: please rename to more EN version `reportsCreated`, `haveReportsBeenCreated`

File: mate-scripts/src/services/Report.service.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+}
+
+export class ReportService {
+  private shouldOpen = true;
+
+  private shouldShowLogs = false;
+
+  private isReporterStarted = false;
+
+  private readonly reportsDir = path.join(this.rootDir, 'reports');
```
Comment (mgrinko) [inline mate-scripts/src/services/Report.service.ts:18]: it is really confusing for me to see that `this` keywords works correctly ouside of a function!

It is a TS magic :)

**Review by mgrinko — APPROVED** [review]

File: mate-scripts/src/services/Report.service.ts (hunk 1 above)
Comment (sergii-nosachenko) [reply mate-scripts/src/services/Report.service.ts:141]: @mgrinko It is a correct executable name: https://www.npmjs.com/package/mochawesome-report-generator

**Pull request merged**